### PR TITLE
fix: bad authorization format crash with 500

### PR DESCRIPTION
**Pull request by zhongwencool**

### CODE REVIEW

File: apps/emqx_management/src/emqx_mgmt_http.erl [hunk 1]
```diff
@@ -118,9 +118,10 @@ handle_request(_Method, _Path, Req) ->
     cowboy_req:reply(400, #{<<"content-type">> => <<"text/plain">>}, <<"Not found.">>, Req).
 
 authorize_appid(Req) ->
-    case cowboy_req:parse_header(<<"authorization">>, Req) of
+    try cowboy_req:parse_header(<<"authorization">>, Req) of
```
Comment (zmstone) [inline apps/emqx_management/src/emqx_mgmt_http.erl:121]:
```suggestion
    try
        case cowboy_req:parse_header(<<"authorization">>, Req) of
```

File: lib-ce/emqx_dashboard/src/emqx_dashboard.erl [hunk 2]
```diff
@@ -103,7 +103,7 @@ is_authorized(Req) ->
 is_authorized("/api/v4/auth", _Req) ->
     true;
 is_authorized(_Path, Req) ->
-    case cowboy_req:parse_header(<<"authorization">>, Req) of
+    try cowboy_req:parse_header(<<"authorization">>, Req) of
```
Comment (zmstone) [inline lib-ce/emqx_dashboard/src/emqx_dashboard.erl:106]:
```suggestion
    try
        case cowboy_req:parse_header(<<"authorization">>, Req) of
```

**Comment by zmstone** [comment]
`try EXPR of`
NOTE: only catches exceptions in `EXPR`, but not the exceptions of unmatched clauses.

always use
```
try
    case EXPR of
```
to be on the safe side.

**Comment by zhongwencool** [comment]
I use `_ -> false `  to aovid unmatched clauses.
```erlang
try cowboy_req:parse_header(<<"authorization">>, Req) of
        {basic, AppId, AppSecret} -> emqx_mgmt_auth:is_authorized(AppId, AppSecret);
         _  -> false
    catch _:_ -> false
    end.
```
the `case` don't look concise at all.
```erlang
try
       case cowboy_req:parse_header(<<"authorization">>, Req) of
            {basic, AppId, AppSecret} ->
                emqx_mgmt_auth:is_authorized(AppId, AppSecret);
            _  ->
                false
        end
    catch _:_ -> false
    end.
```
Now I've changed it to look like this:
```erlang
try
     {basic, AppId, AppSecret} = cowboy_req:parse_header(<<"authorization">>, Req),
     emqx_mgmt_auth:is_authorized(AppId, AppSecret)
 catch _:_ -> false
 end.
```

**Review by HJianBo — APPROVED** [review]

**Pull request merged**